consumer/pdata/trace.go

```diff
@@ -40,15 +40,25 @@ func TracesToOtlp(td Traces) []*otlptrace.ResourceSpans {
 	return *td.orig
 }
 
-// ToOtlpProtoBytes returns the internal Traces to OTLP Collector
-// ExportTraceServiceRequest ProtoBuf bytes. This is intended to export OTLP
-// Protobuf bytes for OTLP/HTTP transports.
+// ToOtlpProtoBytes converts the internal Traces to OTLP Collector
+// ExportTraceServiceRequest ProtoBuf bytes.
 func (td Traces) ToOtlpProtoBytes() ([]byte, error) {
 	return proto.Marshal(&otlpcollectortrace.ExportTraceServiceRequest{
 		ResourceSpans: *td.orig,
 	})
 }
 
+// FromOtlpProtoBytes converts OTLP Collector ExportTraceServiceRequest
+// ProtoBuf bytes to the internal Traces. Overrides current data.
+func (td Traces) FromOtlpProtoBytes(data []byte) error {
+	traces := &otlpcollectortrace.ExportTraceServiceRequest{}
+	if err := proto.Unmarshal(data, traces); err != nil {
+		return err
+	}
+	*td.orig = traces.ResourceSpans
```

Comment:
If you look at similar ToOtlpProtoBytes - it has the same problem (actually I see no problem, no one guarantees zero-initialized structure to work properly). I cannot change that function, because it breaks compatibility, and in my opinion it's better to be consistent in this case. Or even return to the first solution where function wasn't on the structure

Reply:
Hmm, it is unfortunate that we used value receivers on `Traces`, but it is too late now, so you are right, we need to use value receivers for consistency now. Please add a comment to FromOtlpProtoBytes to tell that NewTraces must be used to create the instance before calling this function. It is easy to make the mistake since one will probably try this first and will fail:
```go
var td pdata.Traces
td.FromOtlpProtoBytes(someData)
```